Tests/NIOTransportServicesTests/NIOTSBootstrapTests.swift

```diff
@@ -371,6 +371,87 @@ final class NIOTSBootstrapTests: XCTestCase {
         XCTAssertEqual(try listenerChannel.getOption(NIOTSChannelOptions.multipathServiceType).wait(), .handover)
         XCTAssertEqual(try connectionChannel.getOption(NIOTSChannelOptions.multipathServiceType).wait(), .handover)
     }
+
+    func testNWParametersConfigurator_ListenerUsesChildConfigurator() async throws {
+        try await withEventLoopGroup { group in
+            let configuratorListenerCounter = NIOLockedValueBox(0)
+            let configuratorConnectionCounter = NIOLockedValueBox(0)
+            let waitForConnectionHandler = WaitForConnectionHandler(
+                connectionPromise: group.next().makePromise()
+            )
+
+            let listenerChannel = try await NIOTSListenerBootstrap(group: group)
+                .childChannelInitializer { connectionChannel in
+                    connectionChannel.eventLoop.makeCompletedFuture {
+                        try connectionChannel.pipeline.syncOperations.addHandler(waitForConnectionHandler)
+                    }
+                }
+                .configureNWParameters { _ in
+                    configuratorListenerCounter.withLockedValue { $0 += 1 }
+                }
+                .configureChildNWParameters { _ in
+                    configuratorConnectionCounter.withLockedValue { $0 += 1 }
+                }
+                .bind(host: "localhost", port: 0)
+                .get()
+
+            let connectionChannel: Channel = try await NIOTSConnectionBootstrap(group: group)
+                .connect(to: listenerChannel.localAddress!)
+                .get()
+
+            // Wait for the server to activate the connection channel to the client.
+            try await waitForConnectionHandler.connectionPromise.futureResult.get()
+
+            try await listenerChannel.close().get()
+            try await connectionChannel.close().get()
+
+            XCTAssertEqual(1, configuratorListenerCounter.withLockedValue { $0 })
+            XCTAssertEqual(1, configuratorConnectionCounter.withLockedValue { $0 })
+        }
+    }
+
+    func testNWParametersConfigurator_ClientUsesConfigurator() async throws {
```

Comment:
Is there any reason we shouldn't just test the client in the test above as well? The code is effectively the same. Or separate out the common logic into a helper  method. Either way we should de-duplicate this.

Reply:
Hm yeah I was trying to make it clear that we were testing the client in one and the server on the other, but I merged them and renamed the counters and I think it's clear enough.